### PR TITLE
KMA-11 Develop common overlay

### PR DESCRIPTION
**Pull request by GeorgeKandamkolathy**
Created overlay component with button, title, and close icon.

### CODE REVIEW

File: src/packages/common/overlay/index.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    onClosePress,
+    buttonProps,
+    button,
+    children,
+    ...others
+  } = props;
+
+  return (
+    <ThemeProvider>
+      <EOverlay isVisible={true} overlayStyle={styles.overlay} {...others}>
```
Comment (ericgmoon) [inline src/packages/common/overlay/index.js:24]: `isVisible` automatically defaults to `isVisible={true}`. This is what is causing the linter to fail.

File: src/packages/common/overlay/index.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        <View style={styles.flexContainer}>
+          <View style={styles.flexTitleContainer}>
+            <View>
+              {title ? <Text style={styles.title}>{title}</Text> : null}
+            </View>
+            <View style={{ height: 15, width: 25 }}>
```
Comment (ericgmoon) [inline src/packages/common/overlay/index.js:30]: Could we put this style in `styles.js`?

File: src/packages/common/overlay/styles.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  flexContainer: {
+    display: "flex",
+    flexDirection: "column",
+  },
+  flexTitleContainer: {
+    flex: 1,
+    flexDirection: "row",
+    justifyContent: "space-between",
+    marginBottom: 20,
+    height: 25,
```
Comment (ericgmoon) [inline src/packages/common/overlay/styles.js:17]: Should ideally be a multiple of 4 @m-lo

**Review by ericgmoon — CHANGES_REQUESTED** [review]
Looks mostly great! Refer to my comments for feedback, and tests should be added before the PR can be merged in.

File: src/packages/common/overlay/index.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    close,
+    onClosePress,
+    buttonProps,
+    button,
+    children,
+    ...others
+  } = props;
+
+  return (
+    <ThemeProvider>
```
Comment (nzh20) [inline src/packages/common/overlay/index.js:23]: Why do we have theme provider but no theme prop passed in? Is it being used here then?

Comment (ericgmoon) [reply src/packages/common/overlay/index.js:23]: It is not - I don't believe we need it. I agree it should be removed.

File: package.json [hunk 5]
```diff
@@ -27,9 +27,10 @@
     "react-native": "https://github.com/expo/react-native/archive/sdk-40.0.1.tar.gz",
     "react-native-elements": "^3.1.0",
     "react-native-vector-icons": "^8.0.0",
```
Comment (ericgmoon) [inline package.json:29]: Let's get rid of all changes to `package.json` past line 29.

File: src/packages/common/overlay/index.js [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+              <Icon
+                name="clear"
+                color="black"
+                onPress={onClosePress}
+                type="material"
+              />
+            ) : null}
+          </View>
+        </View>
+
```
Comment (ericgmoon) [inline src/packages/common/overlay/index.js:36]: Get rid of newline here

**Review by ericgmoon — CHANGES_REQUESTED** [review]

**Review by ericgmoon — APPROVED** [review]
Great work! I have checked the visual renders on this manually and it looks good :)

We may need to update it so that the number of actions buttons is customisable, but happy to merge for now.

**Pull request merged**